src/main/java/gregtech/client/model/modelfactories/MaterialBlockBakedModel.java

```diff
@@ -0,0 +1,165 @@
+        this.iconSet = iconSet;
+        this.iconType = iconType;
+
+        this.bakedModelId = new ModelResourceLocation(new ResourceLocation(GTValues.MODID,
+                "material_block_" + iconSet.name + "_" + iconType.name), "");
+    }
+
+    public ModelResourceLocation getBakedModelId() {
+        return bakedModelId;
+    }
+
+    public ResourceLocation getBaseModelLocation() {
+        return this.iconType.getBlockModelPath(this.iconSet);
+    }
+
+    public ModelResourceLocation getModelLocation(String variant) {
+        return new ModelResourceLocation(getBaseModelLocation(), variant);
+    }
+
+    protected IBakedModel getModel() {
+        return Minecraft.getMinecraft().blockRenderDispatcher.getBlockModelShapes().getModelManager()
+                .getModel(getModelLocation(""));
+    }
+
+    @Override
+    public List<BakedQuad> getQuads(@Nullable IBlockState state, @Nullable EnumFacing side, long rand) {
+        return getModel().getQuads(state, side, rand);
+    }
+
+    @Override
+    public boolean isAmbientOcclusion() {
+        return getModel().isAmbientOcclusion();
+    }
+
+    @Override
+    public boolean isGui3d() {
+        return getModel().isGui3d();
+    }
+
+    @Override
+    public boolean isBuiltInRenderer() {
+        return getModel().isBuiltInRenderer();
+    }
+
+    @Override
+    public TextureAtlasSprite getParticleTexture() {
+        return getModel().getParticleTexture();
+    }
+
+    @Override
+    public ItemOverrideList getOverrides() {
+        return getModel().getOverrides();
+    }
+
+    @SuppressWarnings("deprecation")
+    @Override
+    public ItemCameraTransforms getItemCameraTransforms() {
+        return getModel().getItemCameraTransforms();
+    }
+
+    @Override
+    public boolean isAmbientOcclusion(IBlockState state) {
+        return getModel().isAmbientOcclusion(state);
+    }
+
+    @Override
+    public Pair<? extends IBakedModel, Matrix4f> handlePerspective(ItemCameraTransforms.TransformType cameraTransformType) {
+        return getModel().handlePerspective(cameraTransformType);
+    }
+
+    @SubscribeEvent
+    public static void onTextureStitch(TextureStitchEvent.Pre event) {
+        UNBAKED_MODEL_CACHE.clear(); // in case of state desync
+        for (MaterialBlockBakedModel model : INSTANCES.values()) {
+            IModel baseModel;
+            boolean error = false;
+            try {
+                baseModel = ModelLoaderRegistry.getModel(model.getBaseModelLocation());
+            } catch (Exception e) {
+                GTLog.logger.error("Failed to load material model {}:", model.getBaseModelLocation(), e);
+                baseModel = ModelLoaderRegistry.getMissingModel();
+                error = true;
+            }
+
+            if (!error) {
+                for (ResourceLocation texture : baseModel.getTextures()) {
+                    event.getMap().registerSprite(texture);
```

Comment:
Does anything need to be done here to prevent registering duplicate textures?

Reply:
Duplicate texture registration produces no effect.